logbook-jaxrs/src/main/java/org/zalando/logbook/jaxrs/LogbookClientFilter.java

```diff
@@ -37,11 +37,10 @@ public void filter(final ClientRequestContext context) throws IOException {
         if (context.hasEntity()) {
             context.setProperty("request", request);
             context.setProperty("write-request", stage);
+            request.expose();
```

Comment:
I'm a bit puzzled why we'd only do with conditional buffering to a request here and to a response in the server filter? Shouldn't the same check be done on line 61 here
https://github.com/zalando/logbook/blob/c9e8ba9922e6f4e46afb487c4e2ac9ea69b4212d/logbook-jaxrs/src/main/java/org/zalando/logbook/jaxrs/LogbookClientFilter.java#L61

And line 35 in LogbookServerFilter?
https://github.com/zalando/logbook/blob/c9e8ba9922e6f4e46afb487c4e2ac9ea69b4212d/logbook-jaxrs/src/main/java/org/zalando/logbook/jaxrs/LogbookServerFilter.java#L35

Reply:
You have a point.
The main issue is that the tests do not reflect what Quarkus does in reality. 

Debugging the tests, I noted that the default implementation of `State#buffer` is executed, which doesn't do anything: 

https://github.com/zalando/logbook/blob/05ede67de9953342d6fb8992a56e73b02915dba7/logbook-jaxrs/src/main/java/org/zalando/logbook/jaxrs/LocalResponse.java#L31-L33

But running with live Quarkus, the overridden implementation is picked, and that's the source of the NPE:

https://github.com/zalando/logbook/blob/05ede67de9953342d6fb8992a56e73b02915dba7/logbook-jaxrs/src/main/java/org/zalando/logbook/jaxrs/LocalResponse.java#L58-L62

I just found two examples of `expose` and fixed them, assuming they are the only ones. My real-world test was with `HEAD` so I only tested the "empty response" case.

I tried adding Quarkus machinery to our tests, but they always succeeded by following the default implementation of `State#buffer`.

Reply:
Regarding the default implementation of State#buffer, shouldn't that be changed as soon as strategy.process() is called?
 
https://github.com/zalando/logbook/blob/65a12b97b8538a25b284e9edf107ea547dcb59c8/logbook-core/src/main/java/org/zalando/logbook/core/DefaultLogbook.java#L52

But I also couldn't find a way to break anything with this fix, maybe it's sufficient. 

Alternatively, what if the fix is applied to `TeeOutputStream` instead to tolerate `null` OutputStream parameter? 

Overall I'm good with the fix, just trying to understand how to make it regression tolerant and reproducible.

Reply:
I reverted all my changes, and went with your suggestion to make `TeeOutputStream` robust: It now accepts a nullable stream, and handles null properly.